includes/sanitizers/class-amp-form-sanitizer.php

```diff
@@ -75,7 +75,7 @@ public function sanitize() {
 			$xhr_action = $node->getAttribute( 'action-xhr' );
 
 			// Make HTTP URLs protocol-less, since HTTPS is required for forms.
-			if ( 'http://' === strtolower( substr( $action_url, 0, 7 ) ) ) {
+			if ( 0 === stripos( $action_url, 'http://' ) ) {
```

Comment:
See notes above about performance between `strpos()` vs `substr()`. It's O(n) vs O(1).